ci_tools/check_slots.py

```diff
@@ -0,0 +1,109 @@
+""" Script to check that Pyccel coding conventions are correctly followed in the AST
+"""
+import argparse
+import importlib
+import inspect
+import os
+import sys
+from pyccel import ast
+from pyccel.ast.basic import Basic, PyccelAstNode, ScopedNode
+
+parser = argparse.ArgumentParser(description='Check that all new lines in the python files in the pyccel/ code folder are used in the tests')
+parser.add_argument('output', metavar='output', type=str,
+                        help='File where the markdown output will be printed')
+
+args = parser.parse_args()
+
+# Get ast modules
+ast_folder = os.path.dirname(ast.__file__)
+ast_modules = [mod[:-3] for mod in os.listdir(ast_folder) if mod != '__init__.py' and mod.endswith('.py')]
```

Comment:
why is the slots check restricted to the ast modules?

Reply:
I agree with @EmilyBourne